pkg/manager/member/pump_member_manager.go

```diff
@@ -386,32 +386,22 @@ func getNewPumpStatefulSet(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (*app
 		},
 	}
 
+	// TODO: set serviceAccountName in BuildPodSpec
 	serviceAccountName := tc.Spec.Pump.ServiceAccount
 	if serviceAccountName == "" {
 		serviceAccountName = tc.Spec.ServiceAccount
 	}
+	podSpec := spec.BuildPodSpec()
+	podSpec.Containers = containers
+	podSpec.Volumes = volumes
+	podSpec.ServiceAccountName = serviceAccountName
 
-	// TODO: use spec.BuildPodSpec() to generate pod spec
 	podTemplate := corev1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
 			Annotations: podAnnos,
 			Labels:      pumpLabel,
 		},
-		Spec: corev1.PodSpec{
-			Containers:         containers,
-			ServiceAccountName: serviceAccountName,
-			Volumes:            volumes,
-
-			Affinity:         spec.Affinity(),
-			Tolerations:      spec.Tolerations(),
-			NodeSelector:     spec.NodeSelector(),
-			SchedulerName:    spec.SchedulerName(),
-			SecurityContext:  spec.PodSecurityContext(),
-			HostNetwork:      spec.HostNetwork(),
-			DNSPolicy:        spec.DnsPolicy(),
```

Comment:
`DNSPolicy` and `InitContainers` are not covered in the `buildPodSpec()`.

Reply:
!!!